editor/code/views/info/Info.tsx

```diff
@@ -64,12 +71,23 @@ export function InfoPanel() {
   if (!goals) return null;
 
   return (
-    <FileInfo textDocument={goals.textDocument} position={goals.position}>
-      <Goals goals={goals.goals} />
-      <Program program={goals.program} />
-      <Messages messages={goals.messages} />
-      <ErrorBrowser error={goals.error} />
-    </FileInfo>
+    <div className="info-panel-container">
+      <div className="info-panel">
+        <FileInfo textDocument={goals.textDocument} position={goals.position}>
+          <Goals goals={goals.goals} />
+          <Program program={goals.program} />
+          <Messages messages={goals.messages} />
+        </FileInfo>
+      </div>
+      {!goals.error ? null : (
```

Comment:
Moved the check here because this component is the one layering the others. The `ErrorBrowser` shouldn't be the one deciding whether it's displayed or not imo, but the parent that sets the structure.

Reply:
I did understood that in React it was idiomatic for components to handle these kind of cases returning `null` by themselves, but this looks OK to me too.